packages/auth/src/providers/cognito/types/outputs.ts

```diff
@@ -88,6 +94,21 @@ export type SignInWithCustomSRPAuthOutput = AuthSignInOutput;
  */
 export type SignUpOutput = AuthSignUpOutput<AuthVerifiableAttributeKey>;
 
+export type SignUpPasswordlessWithEmailAndMagicLinkOutput = {
+	isSignUpComplete: boolean;
+	nextStep: ConfirmSignInWithMagicLinkSignUpStep;
+};
+
+export type SignUpPasswordlessWithEmailAndOTPOutput = {
+	isSignUpComplete: boolean;
+	nextStep: ConfirmSignInWithOTPSignUpStep;
+};
+
+export type SignUpPasswordlessWithSMSAndOTPOutput = {
+	isSignUpComplete: boolean;
+	nextStep: ConfirmSignInWithOTPSignUpStep;
+};
```

Comment:
if the outputs are going to end up being discriminated unions, then it makes sense to also define the values for `isSignUpComplete` .e.g :
```suggestion
export type SignUpPasswordlessWithEmailAndOTPOutput = {
	isSignUpComplete: true;
	nextStep: ConfirmSignInWithOTPSignUpStep;
};

export type SignUpPasswordlessWithSMSAndOTPOutput = {
	isSignUpComplete: true;
	nextStep: ConfirmSignInWithOTPSignUpStep;
};
```

Reply:
This suggestion is what i was driving at with https://github.com/aws-amplify/amplify-js/pull/12718/files#r1433384269. @israx This would deviate from the existing Auth output interface's `isSignInComplete`/`isSignUpComplete` fields  correct? On one hand could see an argument for consistency by continuing to use `boolean`, but also don't feel there is much value in providing `boolean` as `isSignUpComplete` when we know it will always be `true`